packages/block-library/src/table/save.js

```diff
@@ -56,11 +60,18 @@ export default function save( { attributes } ) {
 
 	return (
 		<figure>
-			<table className={ classes }>
+			<table className={ classes } aria-labelledby={ hasCaption ? captionId : undefined }>
 				<Section type="head" rows={ head } />
 				<Section type="body" rows={ body } />
 				<Section type="foot" rows={ foot } />
 			</table>
+			{ hasCaption && (
+				<RichText.Content
+					id={ captionId }
```

Comment:
I'm not sure about this approach for setting an id in the `save` function.

One problem might be if a post with an existing table is edited and a new table block added, the id for that new block could match this one (as the ids will have been generated in different editor sessions).

Using the clientId of the block might be a better idea.